postgres/datadog_checks/postgres/connections.py

```diff
@@ -1,14 +1,40 @@
 # (C) Datadog, Inc. 2023-present
 # All rights reserved
 # Licensed under a 3-clause BSD style license (see LICENSE)
+import contextlib
 import datetime
 import inspect
 import threading
-from collections import namedtuple
+import time
+from typing import Callable, Dict
 
 import psycopg2
 
-ConnectionWithTTL = namedtuple("ConnectionWithTTL", "connection deadline")
+
+class ConnectionPoolFullError(Exception):
+    def __init__(self, size, timeout):
+        self.size = size
+        self.timeout = timeout
+        super().__init__()
```

Comment:
Is this necessary for exceptions?

Reply:
oh I guess not, I'm not using any of the base class's attributes. I'll remove it